# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/LIROps.yaml
# Commit: 90fa382b5a20
# Full Hash: 90fa382b5a203d2e28bfa69f29ab8edee150b77b
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2021-12-23 09:34:36
# Regressor Bug: 1642412
# File Overlap Count: 8
# Description:
#   Bug 1642412 - wasm: Optimize ion table.get/set/size. r=lth
#   
#   This commit adds ion fast paths for table.get/set when the table
#   representation is anyref. An unconditional fast path is added for
#   table.size.
# ==============================================================================

diff -r 40f861f62a0f -r 90fa382b5a20 js/src/jit/LIROps.yaml
--- a/js/src/jit/LIROps.yaml	Wed Dec 22 17:36:29 2021 +0000
+++ b/js/src/jit/LIROps.yaml	Wed Dec 22 17:36:30 2021 +0000
@@ -2723,9 +2723,18 @@
   arguments:
     offset: size_t
 
+- name: WasmLoadTableElement
+  result_type: WordSized
+  operands:
+    elements: WordSized
+    index: WordSized
+
 - name: WasmDerivedPointer
   gen_boilerplate: false
 
+- name: WasmDerivedIndexPointer
+  gen_boilerplate: false
+
 - name: WasmStoreRef
   operands:
     tls: WordSized